Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -420,34 +420,40 @@ public final class BuildExecutionContext {
-            // Windows rather than the `lib` directory as on unicies.  The `lib`
-            // directory contains the import library (and possibly static
-            // archives) which are used for linking.  The runtime component is
-            // not (necessarily) part of the SDK distributions.
-            //
-            // NOTE: the library name here `libIndexStore.dll` is technically
-            // incorrect as per the Windows naming convention.  However, the
-            // library is currently installed as `libIndexStore.dll` rather than
-            // `IndexStore.dll`.  In the future, this may require a fallback
-            // search, preferring `IndexStore.dll` over `libIndexStore.dll`.
-            let indexStoreLib = buildParameters.toolchain.swiftCompilerPath
-                                    .parentDirectory
-                                    .appending(component: "libIndexStore.dll")
-#else
-            let ext = buildParameters.hostTriple.dynamicLibraryExtension
-            let indexStoreLib = try buildParameters.toolchain.toolchainLibDir.appending(component: "libIndexStore" + ext)
-#endif
-            return try IndexStoreAPI(dylib: indexStoreLib)
+    // MARK: - Private
+
+    private var indexStoreAPICache = ThreadSafeBox<Result<IndexStoreAPI, Error>>()
+
+    /// Reference to the index store API.
+    var indexStoreAPI: Result<IndexStoreAPI, Error> {
+        self.indexStoreAPICache.memoize {
+            do {
+                #if os(Windows)
+                // The library's runtime component is in the `bin` directory on
+                // Windows rather than the `lib` directory as on unicies.  The `lib`
+                // directory contains the import library (and possibly static
+                // archives) which are used for linking.  The runtime component is
+                // not (necessarily) part of the SDK distributions.
+                //
+                // NOTE: the library name here `libIndexStore.dll` is technically
+                // incorrect as per the Windows naming convention.  However, the
+                // library is currently installed as `libIndexStore.dll` rather than
+                // `IndexStore.dll`.  In the future, this may require a fallback
+                // search, preferring `IndexStore.dll` over `libIndexStore.dll`.
+                let indexStoreLib = buildParameters.toolchain.swiftCompilerPath
+                    .parentDirectory
+                    .appending(component: "libIndexStore.dll")
+                #else
+                let ext = buildParameters.hostTriple.dynamicLibraryExtension
+                let indexStoreLib = try buildParameters.toolchain.toolchainLibDir
+                    .appending(component: "libIndexStore" + ext)
+                #endif
+                return try .success(IndexStoreAPI(dylib: indexStoreLib))
+            } catch {
+                return .failure(error)
+            }
```

Comment:
this is the actually change, the rest are formatting (running formatter automatically on all changed files)